tools/reviewer/job/sync/GitHubClient.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2018 The StartupOS Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.startupos.tools.reviewer.job.sync;
+
+import com.google.protobuf.util.JsonFormat;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.GHUserReq;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.GHUserResp;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.GHPullRequestReq;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.GHPullRequestResp;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.GHCommentsReq;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.GHCommentsResp;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.GHPullRequestFilesReq;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.GHPullRequestFilesResp;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.CreateGHPullRequestResp;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.CreateGHPullRequestCommentReq;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.CreateGHPullRequestReq;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.CreateGHReviewCommentReq;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.GHPullRequestsResp;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.GHPullRequestsReq;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Base64;
+
+import static java.net.HttpURLConnection.HTTP_CREATED;
+import static java.net.HttpURLConnection.HTTP_OK;
+
+public class GitHubClient {
+  private static final String BASE_PATH = "https://api.github.com/";
+  private static final String GET_SINGLE_PULL_REQUEST = "repos/%s/pulls/%d";
+  private static final String GET_SINGLE_USER = "users/%s";
+  private static final String CREATE_REVIEW_COMMENT_OR_GET_LIST_REVIEW_COMMENTS_ON_PULL_REQUEST =
```

Comment:
Let's separate these - 
`CREATE_REVIEW_COMMENT_OR_GET_LIST_REVIEW_COMMENTS_ON_PULL_REQUEST`
into
`CREATE_REVIEW_COMMENT_ON_PULL_REQUEST`
`GET_REVIEW_COMMENTS_ON_PULL_REQUEST`
It's ok they have the same value, it's more readable that way.

Same for `CREATE_COMMENT_ON_ISSUE_OR_GET_LIST_COMMENTS_ON_ISSUE`:
`CREATE_COMMENT_ON_ISSUE`
`GET_COMMENTS_ON_ISSUE`

Reply:
Done.